networkmanager/networkmanager_test.go

```diff
@@ -71,542 +120,924 @@ func TestMain(m *testing.M) {
-			spec.Linux.Namespaces[i].Path = networkmanager.GetNetNsPathByName(containerID)
-		}
-	}
+func (c *testCNIInterface) CheckNetwork(ctx context.Context, net *cni.NetworkConfig, rt *cni.RuntimeConf) error {
+	return nil
+}
+
+func (c *testCNIInterface) CheckNetworkList(
+	ctx context.Context, net *cni.NetworkConfigList, rt *cni.RuntimeConf,
+) error {
+	return nil
+}
+
+func (c *testCNIInterface) DelNetwork(ctx context.Context, net *cni.NetworkConfig, rt *cni.RuntimeConf) error {
+	return nil
+}
+
+func (c *testCNIInterface) GetNetworkCachedConfig(net *cni.NetworkConfig, rt *cni.RuntimeConf) (
+	[]byte, *cni.RuntimeConf, error,
+) {
+	return nil, nil, nil
+}
+
+func (c *testCNIInterface) GetNetworkCachedResult(net *cni.NetworkConfig, rt *cni.RuntimeConf) (types.Result, error) {
+	return nil, nil
+}
+
+func (c *testCNIInterface) GetNetworkListCachedResult(
+	net *cni.NetworkConfigList, rt *cni.RuntimeConf,
+) (types.Result, error) {
+	return nil, nil
+}
+
+func (c *testCNIInterface) ValidateNetwork(ctx context.Context, net *cni.NetworkConfig) ([]string, error) {
+	return nil, nil
+}
+
+func (iptables *testIPTablesInterface) isSamePeriod(trafficPeriod int, t1, t2 time.Time) bool {
+	return iptables.disableResetMonitoringTraffic
+}
 
-	for _, mount := range []string{"/bin", "/sbin", "/lib", "/lib64", "/usr", "/etc/nsswitch.conf"} {
-		spec.Mounts = append(spec.Mounts, runtimespec.Mount{
-			Destination: mount,
-			Type:        "bind", Source: mount, Options: []string{"bind", "ro"},
-		})
+func (iptables *testIPTablesInterface) Clear() {
+	for key := range iptables.chain {
+		delete(iptables.chain, key)
 	}
+}
 
-	for _, mount := range []string{"hosts", "resolv.conf"} {
-		spec.Mounts = append(spec.Mounts, runtimespec.Mount{
-			Destination: path.Join("/etc", mount),
-			Type:        "bind", Source: path.Join(imagePath, "etc", mount), Options: []string{"bind", "ro"},
-		})
+func (iptables *testIPTablesInterface) ListWithCounters(table, chain string) ([]string, error) {
+	iptablesData, ok := iptables.chain[chain]
+	if !ok {
+		return nil, networkmanager.ErrRuleNotExist
 	}
 
-	spec.Process.Capabilities.Bounding = append(spec.Process.Capabilities.Bounding, "CAP_NET_RAW")
-	spec.Process.Capabilities.Effective = append(spec.Process.Capabilities.Effective, "CAP_NET_RAW")
-	spec.Process.Capabilities.Inheritable = append(spec.Process.Capabilities.Inheritable, "CAP_NET_RAW")
-	spec.Process.Capabilities.Permitted = append(spec.Process.Capabilities.Permitted, "CAP_NET_RAW")
-	spec.Process.Capabilities.Ambient = append(spec.Process.Capabilities.Ambient, "CAP_NET_RAW")
+	var counters []string
 
-	if specJSON, err = json.Marshal(&spec); err != nil {
-		return aoserrors.Wrap(err)
+	for i := 0; i < iptablesData.countChain; i++ {
+		counters = append(counters, fmt.Sprintf("-c 0 %d", iptablesData.limit))
 	}
 
-	if err = ioutil.WriteFile(path.Join(imagePath, "config.json"), specJSON, 0644); err != nil {
-		return aoserrors.Wrap(err)
+	iptablesData.limit += 20
```

Comment:
Do not add unit test-specific functionality to the implementation of the interface. Add separate method to set data etc. Otherwise, it is hard to read, and hard to use for further tests.

Reply:
use trafficLimitCounter variable instead of hardcoded value